panic: assignment to entry in nil map

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/nuclei/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #nuclei channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### Nuclei version: v3.1.10

### Current Behavior:

Nuclei panic with the following stacktrace:

```
/go/pkg/mod/[github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/request.go:155](http://github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/request.go:155) +0x111
created by [github.com/projectdiscovery/nuclei/v3/pkg/protocols/http](http://github.com/projectdiscovery/nuclei/v3/pkg/protocols/http).(*Request).executeParallelHTTP in goroutine 28259
/go/pkg/mod/[github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/request.go:161](http://github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/request.go:161) +0x138
[github.com/projectdiscovery/nuclei/v3/pkg/protocols/http](http://github.com/projectdiscovery/nuclei/v3/pkg/protocols/http).(*Request).executeParallelHTTP.func1(0xc003a05a40)
/go/pkg/mod/[github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/request.go:732](http://github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/request.go:732) +0x1f97
[github.com/projectdiscovery/nuclei/v3/pkg/protocols/http](http://github.com/projectdiscovery/nuclei/v3/pkg/protocols/http).(*Request).executeRequest(0xc002c49c00, 0xc0053b32f0, 0xc003a05a40, 0xc005917f78, 0x0, 0xc004e8a0c0, 0x0)
/go/pkg/mod/[github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/httputils/chain.go:119](http://github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/httputils/chain.go:119) +0xde
[github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httputils](http://github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httputils).(*ResponseChain).Fill(0xc004e74bd0)
/go/pkg/mod/[github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/httputils/normalization.go:44](http://github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols/http/httputils/normalization.go:44) +0x27e
[github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httputils.readNNormalizeRespBody](http://github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httputils.readNNormalizeRespBody)(0xc005f2cf30?, 0xc00a2ff6b0)
/usr/local/go/src/net/http/header.go:40
net/http.Header.Set(...)
/usr/local/go/src/net/textproto/header.go:22
net/textproto.MIMEHeader.Set(...)
goroutine 28260 [running]:
panic: assignment to entry in nil map
```

--- a/pkg/protocols/http/httputils/normalization.go
+++ b/pkg/protocols/http/httputils/normalization.go
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"compress/gzip"
 	"compress/zlib"
+	"fmt"
 	"io"
 	"net/http"
 	"strings"
@@ -19,11 +20,18 @@ import (
 // and fills body buffer with actual response body.
 func readNNormalizeRespBody(rc *ResponseChain, body *bytes.Buffer) (err error) {
 	response := rc.resp
+	if response == nil {
+		return fmt.Errorf("something went wrong response is nil")
+	}
 	// net/http doesn't automatically decompress the response body if an
 	// encoding has been specified by the user in the request so in case we have to
 	// manually do it.
 
 	origBody := rc.resp.Body
+	if origBody == nil {
+		// skip normalization if body is nil
+		return nil
+	}
 	// wrap with decode if applicable
 	wrapped, err := wrapDecodeReader(response)
 	if err != nil {
@@ -41,6 +49,9 @@ func readNNormalizeRespBody(rc *ResponseChain, body *bytes.Buffer) (err error) {
 		}
 		if stringsutil.ContainsAny(err.Error(), "unexpected EOF", "read: connection reset by peer", "user canceled") {
 			// keep partial body and continue (skip error) (add meta header in response for debugging)
+			if response.Header == nil {
+				response.Header = make(http.Header)
+			}
 			response.Header.Set("x-nuclei-ignore-error", err.Error())
 			return nil
 		}

